internal/precache/precache.go

```diff
@@ -167,28 +173,28 @@ func QueryJobStatus(ctx context.Context, c client.Client) (*Status, error) {
 			strProgress := strings.TrimSpace(string(data))
 			err = json.Unmarshal([]byte(strProgress), &status.Progress)
 			if err != nil {
-				log.Error(err, "Failed to parse progress", "StatusFile", StatusFile)
+				handler.Log.Error(err, "Failed to parse progress", "StatusFile", StatusFile)
 			} else {
 				status.Message = fmt.Sprintf("total: %d (pulled: %d, skipped: %d, failed: %d)",
 					status.Progress.Total, status.Progress.Pulled, status.Progress.Skipped, status.Progress.Failed)
 			}
 		} else {
-			log.Info("Unable to read precaching progress file", "StatusFile", StatusFile)
+			handler.Log.Info("Unable to read precaching progress file", "StatusFile", StatusFile)
 		}
 	}
 	return status, nil
 }
 
 // Cleanup deletes the ConfigMap and Job precaching resources
-func Cleanup(ctx context.Context, c client.Client) error {
+func (handler *PHandler) Cleanup(ctx context.Context) error {
 	// Delete Job
-	if err := deleteJob(ctx, c, LcaPrecacheJobName, LcaPrecacheNamespace); err != nil {
-		log.Error(err, "Failed to delete precaching job", "name", LcaPrecacheJobName)
+	if err := deleteJob(ctx, handler.Client, LcaPrecacheJobName, LcaPrecacheNamespace); err != nil {
+		handler.Log.Error(err, "Failed to delete precaching job", "name", LcaPrecacheJobName)
 		return err
 	}
 	// Delete ConfigMap
-	if err := deleteConfigMap(ctx, c, LcaPrecacheConfigMapName, LcaPrecacheNamespace); err != nil {
-		log.Error(err, "Failed to delete precaching configmap", "name", LcaPrecacheConfigMapName)
+	if err := deleteConfigMap(ctx, handler.Client, LcaPrecacheConfigMapName, LcaPrecacheNamespace); err != nil {
+		handler.Log.Error(err, "Failed to delete precaching configmap", "name", LcaPrecacheConfigMapName)
 		return err
```

Comment:
Since we are return the errors with requeueWithError. The error will show up in the pods logs anyway. This means we are showing the same error twice? 

Maybe something like will avoid printing it twice + logging something extra for the user?
```
handler.Log.Error(fmt.Errorf("Failed to delete precaching job"), LcaPrecacheJobName))
return err 
```

Or maybe simpler would be just to remove L197 (and other place were we end with the requeueWithError). wdyt?

Reply:
Great point! I have updated the logging to info and omitted the error message as it will be logged! Thanks Nahian :)